java/client/src/org/openqa/selenium/support/ui/ExpectedConditions.java

```diff
@@ -933,4 +934,613 @@ private static WebElement findElement(By by, WebDriver driver) {
+                                                       final String value) {
+    return new ExpectedCondition<Boolean>() {
+      private String currentValue = null;
+
+      @Override
+      public Boolean apply(WebDriver driver) {
+        WebElement element = driver.findElement(locator);
+        currentValue = element.getAttribute(attribute);
+        if (currentValue == null) {
+          currentValue = element.getCssValue(attribute);
+        }
+        return value.equals(currentValue);
+      }
+
+      @Override
+      public String toString() {
+        return String.format("value to be \"%s\". Current value: \"%s\"", value, currentValue);
+      }
+    };
+  }
+
+  /**
+   * An expectation for checking WebElement with given locator has specific text
+   *
+   * @param locator used to find the element
+   * @param value   used as expected text
+   * @return Boolean true when element has text value equal to @value
+   */
+  public static ExpectedCondition<Boolean> textIs(final By locator, final String value) {
+    return new ExpectedCondition<Boolean>() {
+      private String currentValue = null;
+
+      @Override
+      public Boolean apply(WebDriver driver) {
+        try {
+          currentValue = driver.findElement(locator).getText();
+          return currentValue.equals(value);
+        } catch (Exception e) {
+          return false;
+        }
+      }
+
+      @Override
+      public String toString() {
+        return String.format("text to be \"%s\". Current text: \"%s\"", value, currentValue);
+      }
+    };
+  }
+
+  /**
+   * An expectation for checking WebElement with given locator has text with a value as a part of
+   * it
+   *
+   * @param locator used to find the element
+   * @param pattern   used as expected text matcher pattern
+   * @return Boolean true when element has text value containing @value
+   */
+  public static ExpectedCondition<Boolean> textMatches(final By locator, final Pattern pattern) {
+    return new ExpectedCondition<Boolean>() {
+      private String currentValue = null;
+
+      @Override
+      public Boolean apply(WebDriver driver) {
+        try {
+          currentValue = driver.findElement(locator).getText();
+          return pattern.matcher(currentValue).find();
+        } catch (Exception e) {
+          return false;
+        }
+      }
+
+      @Override
+      public String toString() {
+        return String.format("text to match pattern \"%s\". Current text: \"%s\"", pattern.pattern(), currentValue);
+      }
+    };
+  }
+
+  /**
+   * An expectation for checking number of WebElements with given locator
+   *
+   * @param locator used to find the element
+   * @param number  user to define exact number of elements
+   * @return Boolean true when size of elements list is equal to defined
+   */
+  public static ExpectedCondition<Boolean> numberOfElementsIsMoreThan(final By locator,
```

Comment:
Would seem beneficial for the 'numberOfElements....' methods to return List<WebElement> instead